Sources/HTTP/HTTPServer.swift

```diff
@@ -28,31 +31,72 @@ public class HTTPServer {
     /// To process incoming requests
     public let handler: HTTPRequestHandler
 
-    private let server = PoCSocketSimpleServer()
+    /// Stop the server
+    public func stop() {
+        server?.close()
+    }
+
+    /// The port number the server is listening on
+    public var port: Int {
+        return options.port
+    }
+
+    /// Reference to the server for closing
+    private var server: TCPServer?
+
+    /// The number of current connections
+    public var connectionCount: Int
 
     /// Create an instance of the server. This needs to be followed with a call to `start(port:handler:)`
     public init(with newOptions: Options, requestHandler: @escaping HTTPRequestHandler) {
         options = newOptions
         handler = requestHandler
+        connectionCount = 0
     }
 
     /// Start the HTTP server on the given `port` number, using a `HTTPRequestHandler` to process incoming requests.
     public func start() throws {
-        try server.start(port: options.port, handler: handler)
+        let socket = try TCPSocket(isNonBlocking: true, shouldReuseAddress: true)
+        var eventLoops: [DispatchQueue] = []
+        for i in 1...8 {
+            eventLoops.append(.init(label: "org.swift.server.eventLoop.\(i)"))
+        }
+        let server = TCPServer(socket: socket, eventLoops: eventLoops)
+        print("Starting on http://localhost:\(options.port)/")
+        try server.start(hostname: "0.0.0.0", port: UInt16(options.port), backlog: 4096) { client in
+            self.connectionCount += 1
+            client.onClose = {
+                self.connectionCount -= 1
```

Comment:
Isn't this running in a different thread and needs an atomic guard? 🤔

Reply:
Ah yes. That should be dispatched async to the server's queue. What is the connection count used for btw? Do we need that?

Reply:
Yeah, I think a count is pretty useless. A connection list could be useful for debugging.